magefile.go

```diff
@@ -366,6 +372,85 @@ func (b Build) buildOLMBundleImage(imageCacheDir string) error {
 	return nil
 }
 
+func (b Build) buildPackageOperatorImage(imageCacheDir string) error {
+	mg.Deps(
+		Build.init,
+		Dependency.Kustomize,
+	)
+
+	imageTag := imageURL("addon-operator-package")
+
+	// Create directories
+	manifestsDir := path.Join(imageCacheDir, "manifests")
+	tmpDir := path.Join(imageCacheDir, "tmp")
+	for _, command := range [][]string{
+		{"mkdir", "-p", manifestsDir},
+		{"mkdir", "-p", tmpDir},
+	} {
+		if err := sh.RunV(command[0], command[1:]...); err != nil {
+			return err
+		}
+	}
+
+	deploymentTemplate, err := ioutil.ReadFile("config/package/addon-operator.yaml.tpl")
+	if err != nil {
+		return fmt.Errorf("reading deployment template: %w", err)
+	}
+
+	templatedDeployment, err := replaceManagerContainerImage(deploymentTemplate, imageURL("addon-operator-manager"))
+	if err != nil {
+		return err
+	}
+
+	if err = ioutil.WriteFile(manifestsDir+"/addon-operator.yaml",
+		templatedDeployment, os.ModePerm); err != nil {
+		return err
+	}
```

Comment:
shall we replace this logic with a 
`kustomize build` command.

Then we can keep:
- config/package/addon-operator.yaml.tpl as a `.yaml` file which can be `kubectl apply` ed. 
- as we are using kustomize we can set overlays for local development, olm based bundles, Package-operator etc.

Reply:
Hey Nikhil! Yeah, that is what I wanted to do! I tried for a few hours :sweat_smile: The problem I ran into was I wanted to use `kustomize edit set image` to get the image tag into the kustomize file. However you can't specify a path for `kustomize edit set image` (see https://github.com/kubernetes-sigs/kustomize/issues/2803) so I tried to do it by changing directories. This had some weird results, I think because other mage functions were running at the same time. I thought about specifying the tasks to run serially, but I thought that might would be too error prone. What do you think? Do you know a better way to do it? :)

Reply:
Wait a second, it might work

Reply:
Actually it works with changing the directory of the command but not with `os.Chdir`